core/computed/entity-classification.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {makeComputedArtifact} from './computed-artifact.js';
+import {NetworkRecords} from './network-records.js';
+import {Util} from '../util.cjs';
+import UrlUtils from '../lib/url-utils.js';
+import thirdPartyWeb from '../lib/third-party-web.js';
+
+/** @typedef {Map<string, LH.Artifacts.RecognizableEntity>} EntityCache */
+
+class EntityClassification {
+  /**
+   * @param {EntityCache} entityCache
+   * @param {string} url
+   * @return Entity | undefined
+   */
+  static makeUpAnEntity(entityCache, url) {
+    if (!UrlUtils.isValid(url)) return;
```

Comment:
Given `url` seems to always be passed in from a network record or from the URL artifact, is this not always valid?

Perhaps pass in a real `URL` instance if you want runtime guarantee of its validity. Getting this function to not return `undefined` should make usage cleaner in compute_

Reply:
`NetworkRecords` + `URL.{mainDocumentUrl | finalDisplayedUrl}`. Would there be non-network protocol URIs that we have to deal with (`chrome://`, `file://`, and `data:` for example)? I see [a test](https://github.com/GoogleChrome/lighthouse/blob/aecbb7263ba051c2d3e3119f2b21860f0a15035d/core/test/audits/third-party-summary-test.js#L101) with devtoolsLogs containing `chrome://`.

Reply:
TODO for @alexnj, add `UrlUtils.isNonNetworkProtocol` as in `NetworkRequest.isNonNetworkRequest` check here.